woocommerce/woocommerce-gutenberg-products-block
# Move Checkout block attributes into metadata (`block.json`)

This PR will move the attributes for the Checkout block out of `attributes.ts` and into `block.json` this is done to unify the block attributes which is part of #5103.

I'm making this PR now before proceeding to make sure that the approach is correct before addressing all the other blocks.

### Changes
- Added a `get_block_metadata_path` function to the `Assets\Api` class, this allows us to get the path to a `block.json` file (which will be specified in the block's class).
- Removes `attributes` and `supports` from the `$block_settings` array in `AbstractBlock`, this is required because when registering a block with metadata, the values in the `$block_settings` override it. We want the metadata to be the source of truth to avoid duplicating config in PHP and JS. We need to keep `$block_settings` around though to pass `render_callback`, `editor_script`, `editor_style`, and `style` to `register_block_type`. These
- Moved all static attributes from `attributes.ts` into `block.json` (I left the ones that were derived from `getSetting`).
- Moved all static data from `index.ts` into `block.json`

### Testing

#### Automated Tests

### Manual Testing
1. Smoke test the checkout block, try placing an order, using different payment methods, make sure validation etc. is all working as intended

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

### Changelog
> Skipping
